src/main/java/org/jabref/gui/openoffice/OOBibBase.java

```diff
@@ -235,30 +233,24 @@ public void selectDocument() throws NoDocumentException, NoSuchElementException,
         return result;
     }
 
-    private XDesktop simpleBootstrap(String pathToExecutable)
+    private XDesktop simpleBootstrap(List<URL> jarUrls)
         throws IllegalAccessException, InvocationTargetException, BootstrapException,
-        CreationException, IOException {
+        CreationException, IOException, ClassNotFoundException {
 
-        ClassLoader loader = ClassLoader.getSystemClassLoader();
-        if (loader instanceof URLClassLoader) {
-            URLClassLoader cl = (URLClassLoader) loader;
-            Class<URLClassLoader> sysclass = URLClassLoader.class;
-            try {
-                Method method = sysclass.getDeclaredMethod("addURL", URL.class);
-                method.setAccessible(true);
-                method.invoke(cl, new File(pathToExecutable).toURI().toURL());
-            } catch (SecurityException | NoSuchMethodException | MalformedURLException t) {
-                LOGGER.error("Error, could not add URL to system classloader", t);
-                cl.close();
-                throw new IOException("Error, could not add URL to system classloader", t);
-            }
-        } else {
-            LOGGER.error("Error occured, URLClassLoader expected but " + loader.getClass()
-                         + " received. Could not continue.");
+    
+        URL[] urls = jarUrls.toArray(new URL[3]);
+        Class<Bootstrap> clazz  = (Class<Bootstrap>) Class.forName("com.sun.star.comp.helper.Bootstrap", true, new ChildFirstClassLoader(urls, null));
```

Comment:
I don't really understand the problem, but doesn't https://stackoverflow.com/a/48323948/873661 work?

Reply:
The problem is that by default java delegate class loading to the parent class loader and so on until the class is found. This results in loading the` org.jabref.thirdparty` module from the module path and not the jars from the libre office program folder. Calling Bootstrap will then result in an error message, no office executable found.

That's why I created a child first class loader. It loads the stuff into the unnamed module aka the class path. But casting the loaded class back fails because it's different now twice in the app...
https://stackoverflow.com/questions/54536895/cannot-be-cast-to-class-because-they-are-in-unnamed-module-of-loader-app#comment96026301_54536895

Child First class loader:
https://medium.com/@isuru89/java-a-child-first-class-loader-cbd9c3d0305

Reply:
The only solution I have to try is to load the jars into the module path or to put our patched module into the libre office program folder.

Reply:
Ok, but then why are we trying to load jar files that we already include?

Reply:
We must include them to have the types.
But we need to load them dynamically again from libreoffice program folder because otherwise the soffice.exe can't be found.

Reply:
Sounds weird, but I debugged through it and had these errors. Of course lO was running

Reply:
@tobiasdiez  This is the problem with the Bootstrap:
https://github.com/LibreOffice/core/blob/330df37c7e2af0564bcd2de1f171bed4befcc074/javaunohelper/com/sun/star/comp/helper/Bootstrap.java#L332-L338